custom_components/frigate/image.py

```diff
@@ -0,0 +1,120 @@
+"""Support for Frigate images."""
+from __future__ import annotations
+
+import datetime
+import logging
+from typing import Any
+
+from homeassistant.components.image import Image
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import (
+    FrigateMQTTEntity,
+    ReceiveMessage,
+    get_cameras_and_objects,
+    get_friendly_name,
+    get_frigate_device_identifier,
+    get_frigate_entity_unique_id,
+)
+from .const import ATTR_CONFIG, DOMAIN, NAME
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Image entry setup."""
+
+    frigate_config = hass.data[DOMAIN][entry.entry_id][ATTR_CONFIG]
+
+    async_add_entities(
+        [
+            FrigateMqttSnapshots(entry, frigate_config, cam_name, obj_name)
+            for cam_name, obj_name in get_cameras_and_objects(frigate_config, False)
+        ]
+    )
+
+
+class FrigateMqttSnapshots(FrigateMQTTEntity, Image):  # type: ignore[misc]
+    """Frigate best image class."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        frigate_config: dict[str, Any],
+        cam_name: str,
+        obj_name: str,
+    ) -> None:
+        """Construct a FrigateMqttSnapshots image."""
+        self._frigate_config = frigate_config
+        self._cam_name = cam_name
+        self._obj_name = obj_name
+        self._last_image_timestamp = None
+        self._last_image: bytes | None = None
+
+        FrigateMQTTEntity.__init__(
+            self,
+            config_entry,
+            frigate_config,
+            {
+                "state_topic": {
+                    "msg_callback": self._state_message_received,
+                    "qos": 0,
+                    "topic": (
+                        f"{self._frigate_config['mqtt']['topic_prefix']}"
+                        f"/{self._cam_name}/{self._obj_name}/snapshot"
+                    ),
+                    "encoding": None,
+                },
+            },
+        )
+        Image.__init__(self)
+
+    @callback  # type: ignore[misc]
+    def _state_message_received(self, msg: ReceiveMessage) -> None:
+        """Handle a new received MQTT state message."""
+        self._last_image_timestamp = datetime.datetime.now()
+        self._last_image = msg.payload
+        self.async_write_ha_state()
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID to use for this entity."""
+        return get_frigate_entity_unique_id(
+            self._config_entry.entry_id,
+            "image",
```

Comment:
Maybe "image_best_snapshot" (since we could very well have other images in the future).